### PR TITLE
Add subunit2disk which exports a stream to the fs.

### PR DESCRIPTION
**Pull request by rbtcollins**

### CODE REVIEW

**Comment by rbtcollins** [comment]
Timing out - no reviews.

File: python/subunit/_to_disk.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import sys
+from textwrap import dedent
+
+from testtools import StreamToDict
+
+from subunit.filters import run_tests_from_stream
+
+
+def _allocate_path(root, sub):
+    """Figoure a path for sub under root.
```
Comment (jml) [inline python/subunit/_to_disk.py]: "Figure"

File: python/subunit/_to_disk.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    If sub tries to escape root, squash it with prejuidice.
+
+    If the path already exists, a numeric suffix is appended.
+    E.g. foo, foo-1, foo-2, etc.
+
+    :return: the full path to sub.
+    """
+    # subpathss are allowed, but not parents.
```
Comment (jml) [inline python/subunit/_to_disk.py]: "subpaths"

File: python/subunit/_to_disk.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        details = test_dict['details']
+        status = test_dict['status']
+        start, stop = test_dict['timestamps']
+        test_summary = {}
+        test_summary['id'] = id
+        test_summary['tags'] = tags
+        test_summary['status'] = status
+        test_summary['details'] = sorted(details.keys())
+        test_summary['start'] = _json_time(start)
+        test_summary['stop'] = _json_time(stop)
```
Comment (jml) [inline python/subunit/_to_disk.py]: I'd consider extracting this into a pure function, separate from the IO that follows.

File: python/subunit/_to_disk.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    if stdout is None:
+        stdout = sys.stdout
+    if stdin is None:
+        stdin = sys.stdin
+    parser = optparse.OptionParser(
+        description="Export a subunit stream to files on disk.",
+        epilog=dedent("""\
+            Creates a directory per test id, a JSON file with test
+            metadata within that directory, and each attachment
+            is written to their name relative to that directory.
```
Comment (jml) [inline python/subunit/_to_disk.py]: Ah! I was wondering what this was actually for. This is pretty cool. 

I'm guessing that the main use-case is easily getting to attachments? If so (or indeed, even if not), users will probably be able to find this better if there are example use cases in the documentation.

**Comment by jml** [comment]
Really nice, thanks.

Only nits, and a request for some more documentation.

**Pull request merged**